src/api/core/warren/tests/test_utils.py

```diff
@@ -21,3 +29,164 @@ def mult(x):
     assert pipe(add)(1) == 2
     with pytest.raises(TypeError):
         pipe()()
+
+
+def test_get_token():
+    """Test the get_token function."""
+    # Mock signing env variables.
+    settings.APP_SIGNING_KEY = "SigningKeyToChange__FOR_TEST_ONLY"
+    settings.APP_SIGNING_ALGORITHM = "HS256"
+
+    # Mock lti parameters
+    lti_user = {
+        "platform": "http://fake-lms.com",
+        "course": "course-v1:openfun+mathematics101+session01",
+        "email": "johndoe@example.com",
+        "user": "johndoe",
+    }
+    timestamp = int(datetime.datetime.now().timestamp())
+    lti_parameters = {
+        "token_type": "lti_access",
+        "exp": timestamp + 10000,
+        "iat": timestamp,
+        "jti": "",
+        "session_id": str(uuid.uuid4()),
+        "roles": ["instructor"],
+        "user": lti_user,
+        "locale": "fr",
+        "resource_link_id": "8d5dabc2-6af4-42ac-a29b-97649db4a162",
+        "resource_link_description": "",
+    }
+
+    token = jwt.encode(
+        lti_parameters,
+        settings.APP_SIGNING_KEY,
+        algorithm=settings.APP_SIGNING_ALGORITHM,
+    )
+
+    # Decode token, verify its signature and validate its payload
+    payload = get_token(token)
+
+    assert payload == lti_parameters
+
+
+def test_get_token_invalid_signature():
+    """Test the get_token function with an invalid signature."""
+    # Mock signing env variables.
+    settings.APP_SIGNING_KEY = "SigningKeyToChange__FOR_TEST_ONLY"
+    settings.APP_SIGNING_ALGORITHM = "HS256"
+
+    # Mock lti parameters
+    lti_user = {
+        "platform": "http://fake-lms.com",
+        "course": "course-v1:openfun+mathematics101+session01",
+        "email": "johndoe@example.com",
+        "user": "johndoe",
+    }
+    timestamp = int(datetime.datetime.now().timestamp())
+    lti_parameters = {
+        "token_type": "lti_access",
+        "exp": timestamp + 10000,
+        "iat": timestamp,
+        "jti": "",
+        "session_id": str(uuid.uuid4()),
+        "roles": ["instructor"],
+        "user": lti_user,
+        "locale": "fr",
+        "resource_link_id": "8d5dabc2-6af4-42ac-a29b-97649db4a162",
+        "resource_link_description": "",
+    }
+
+    # A wrong signing key is used to encode the token
+    token = jwt.encode(
+        lti_parameters, "WRONG_KEY", algorithm=settings.APP_SIGNING_ALGORITHM
+    )
+
+    # Token should not be valid
+    with pytest.raises(HTTPException) as exception:
```

Comment:
Two remarks:

1. as the `get_lti_token` utility may be used outside from an HTTP request context, it should raise a more generic error such as a `ValueError`, and this error may be catch in an HTTP request context that raises an `HTTPException`,
2. it's a good idea to also test raised exception message using `match`!

```suggestion
    with pytest.raises(ValueError, match="Could not validate credentials") as exception:
```

Reply:
After thought: forget the first point as it's injected as a dependency of API views.

Reply:
Actually, I was working on this part while you did your review, I have added an assent on the response's detail. WDYT?

Reply:
It's not in a fixup commit, but in the commit itself !